### PR TITLE
slippage setting added to guni

### PR DESCRIPTION
**Pull request by piekczyk**
# [Slippage setting added to guni](https://app.shortcut.com/oazo-apps/story/3140/bug-slippage-wrongly-accounted-for-in-worst-case-market-price-calculation)

<please insert a clubhouse link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>
- slippage setting added to guni
  
## How to test 🧪
  <Please explain how to test your changes>
- heroku / dev
    
## Definition of done ✔️

- [ ] Acceptance criteria for each issue met
- [ ] Unit tests written where needed and passing
- [ ] End-to-end tests for happy path
- [ ] Documentation updated <When applicable>
- [ ] Project builds without errors
- [ ] Non-functional requirements met
- [ ] Code reviewed and functionality tested by the reviewer (locally, Heroku, etc.)
- [ ] Feature verified and accepted by Product Owner
- [ ] Project deployed to production environment

### CODE REVIEW

**Review by adamskrodzki — CHANGES_REQUESTED** [review]
does this change requre corresponding change in Exchange SC ? @fenris85 i believe Georgie changes are not merged

**Comment by fenris85** [comment]
> does this change requre corresponding change in Exchange SC ? @fenris85 i believe Georgie changes are not merged

no, this change is not merged in SC

File: features/openGuniVault/openGuniVault.ts [hunk 1]
```diff
@@ -362,19 +362,38 @@ export function createOpenGuniVault$(
...
+                      withLatestFrom(stateSubjectShared$),
+                      switchMap(([change, state]) => {
                         const depositAmount =
                           change.kind === 'injectStateOverride'
                             ? change.stateToOverride.depositAmount
-                            : change.depositAmount
+                            : change.kind === 'depositAmount'
+                            ? change.depositAmount
+                            : state.depositAmount
+
```
Comment (sennett) [inline features/openGuniVault/openGuniVault.ts:396]: I think it's strange that we are passing the state subject back into the middle of itself.  The state subject is the view state, which is the result of this pipeline, and composed from other streams.  I'm not 100% sure of having a stream (`OpenGuniVaultState`) depend on itself in this way - it seems like a code smell if we are having to do this.

**Review by sennett — COMMENTED** [review]
I'm keen to have a chat about passing the stream back into itself and see if we can find another solution.

File: features/openGuniVault/openGuniVault.ts (hunk 1 above)
Comment (piekczyk) [reply features/openGuniVault/openGuniVault.ts:396]: Yep, it's pretty hacky, but for now we couldn't find better way to fix this problem. Other pipelines don't have this issue.

File: features/openGuniVault/openGuniVault.ts [hunk 2]
```diff
@@ -272,15 +272,16 @@ export function createOpenGuniVault$(
     amountOMax: BigNumber
     amount1Max: BigNumber
   }) => Observable<{ amount0: BigNumber; amount1: BigNumber; mintAmount: BigNumber }>,
+  slippageLimit$: Observable<UserSettingsState>,
 ): Observable<OpenGuniVaultState> {
   return ilks$.pipe(
```
Comment (adamskrodzki) [inline features/openGuniVault/openGuniVault.ts:277]: I have a feeling like we could Use some tests of this pipeline modification,
some similar tests already exists for other flows
features/openMultiplyVault/tests/openMultiplyVault.test.ts

In perfect scenario there woould be separate PR with test of slippage value change impact on Guni pipeline

**Review by adamskrodzki — CHANGES_REQUESTED** [review]
Is there possibility to split this PR to two:

One responsible for UI part - passing slippage to a user and second for fixing calculaton, I think it is important from testability perspective. 

Not a requirement (since i was creating similar PR's myself :) ), but would help to review

File: features/manageGuniVault/manageGuniVault.ts [hunk 3]
```diff
@@ -260,50 +262,56 @@ export function createManageGuniVault$(
...
-
-                          return {
-                            kind: 'guniTxData',
-                            swap,
-                            sharedAmount0,
-                            sharedAmount1: sharedAmount1.minus(0.01),
-                            requiredDebt,
-                            fromTokenAmount: swap.collateralAmount,
-                            toTokenAmount: swap.daiAmount,
-                            minToTokenAmount: swap.daiAmount.times(one.minus(SLIPPAGE)),
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply features/manageGuniVault/manageGuniVault.ts:287]: sorry for that, the update related to market price max slippage is in following commit:

https://github.com/OasisDEX/oasis-borrow/pull/638/commits/5b03a1fc020328a5808533ec5b3090400a489d9e

File: features/openGuniVault/openGuniVault.ts (hunk 2 above)
Comment (piekczyk) [reply features/openGuniVault/openGuniVault.ts:277]: Yep, we currently don't have tests for guni and I believe question if we should prepare these test before / after refactor has been left without answer / decision. Anyway this PR will be merge to feature branch so it won't ruin staging.

**Review by sennett — APPROVED** [review]
Seems good to me - it's a shame about the coverage decreasing and the state getting passed back in to the stream, but I don't know enough yet to offer a better solution quickly.

**Pull request merged**